### ISSUE TITLE
UnusedArgumentsRule should check if method is an implementation of an event

### ISSUE THREAD

**Issue by crenan** [JAM-Software/SonarDelphi#26]
I have a Code Smell "_Unused argument: AMyUnusedParam at TFormTest.MyComponentMyEventImplementation_" for the example below

``` Delphi
procedure TFormTest.MyComponentMyEventImplementation(AMyUnusedParam: TSomeRandomType);
begin
  inherited;
{*
  Some code that doesn't use "AMyUnusedParam"
*}
end;
```

I think this rule should check if the method is a implementation of an event before mark the params as code smells.
Generally the code in events not necessarly need to use all the parameters as they exists only to comply with the event declaration.

**Comment by joachimmarder** [comment]
I don't think that this is possible. You cannot check generally if a method might get assigned to an event or not. A similar problem exists for interface implementations, where it is common that not every implementation uses every parameter. But you cannot know if a method fulfills an interface in a derived class.

**Comment by crenan** [comment]
@joachimmarder good point, but it is possible to reduce this false-positive by checking if the method is bound to an event in the included DFM:

- Check if {$R *.DFM} exists
- Open the corresponding *.DFM
- Search for the method name
